#!/usr/bin/env python
PACKAGE = "dynamic_transform_publisher"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frame_id",		str_t,		0, "The name of the frame to reference", "")
gen.add("child_frame_id",	str_t, 		0, "The name of the frame to broadcast", "")
gen.add("period",       double_t,   0, "period (in ms)", 100, 0, 1000)
gen.add("x", 			double_t,	0, "x", 0, -100, 100)
gen.add("y",			double_t,	0, "y", 0, -100, 100)
gen.add("z", 			double_t,	0, "z", 0, -100, 100)
gen.add("use_rpy",		bool_t,		0, "Use RPY values", True)
group = gen.add_group("RPY")
group.add("roll",		double_t,	0, "roll", 0, -3.14, 3.14)
group.add("pitch",		double_t,	0, "pitch", 0, -3.14, 3.14)
group.add("yaw",		double_t,	0, "yaw", 0, -3.14, 3.14)
group = gen.add_group("Quaternion")
group.add("qx",			double_t,	0, "qx", 0, -1, 1)
group.add("qy",			double_t,	0, "qy", 0, -1, 1)
group.add("qz", 		double_t,	0, "qz", 0, -1, 1)
group.add("qw",			double_t,	0, "qw", 1, -1, 1)


exit(gen.generate(PACKAGE, "dynamic_transform_publisher", "TF"))
